### PR TITLE
feat(gatsby-source-wordpress): opt out of stale node checks

### PR DESCRIPTION
**Pull request by TylerBarnes**
This PR adds support for the new `enableStatefulSourceNodes` API I added in https://github.com/gatsbyjs/gatsby/pull/37782

canary available as `gatsby-source-wordpress@7.9.0-stateful-wp.2`

### CODE REVIEW

**Comment by LekoArts** [comment]
Duplicate of https://github.com/gatsbyjs/gatsby/pull/37795 ?

File: packages/gatsby-source-wordpress/src/steps/source-nodes/index.ts [hunk 1]
```diff
@@ -8,9 +8,14 @@ import store from "~/store"
...
 import { allowFileDownloaderProgressBarToClear } from "./create-nodes/create-remote-file-node/progress-bar-promise"
 import { sourcePreviews } from "~/steps/preview"
+import { hasStatefulSourceNodes } from "~/utils/gatsby-features"
 
 const sourceNodes: Step = async helpers => {
-  const { cache, webhookBody, refetchAll } = helpers
+  const { cache, webhookBody, refetchAll, actions } = helpers
+
+  if (hasStatefulSourceNodes) {
+    actions.enableStatefulSourceNodes()
```
Comment (pieh) [inline packages/gatsby-source-wordpress/src/steps/source-nodes/index.ts:17]:
```
packages/gatsby-source-wordpress/src/steps/source-nodes/index.ts:17:13 - error TS2339: Property 'enableStatefulSourceNodes' does not exist on type 'Actions'.

17     actions.enableStatefulSourceNodes()
               ~~~~~~~~~~~~~~~~~~~~~~~~~


Found 1 error in packages/gatsby-source-wordpress/src/steps/source-nodes/index.ts:17
```

Seems like we are also missing some types for this in https://github.com/gatsbyjs/gatsby/blob/ccb0e7a153e0866f3991c25c9d1a27e345c625d1/packages/gatsby/index.d.ts#L1291

It seems like there this action was added wrongly to `GatsbyNode` API list and not `Actions`:

https://github.com/gatsbyjs/gatsby/blob/ccb0e7a153e0866f3991c25c9d1a27e345c625d1/packages/gatsby/index.d.ts#L437-L440

Comment (pieh) [reply packages/gatsby-source-wordpress/src/steps/source-nodes/index.ts:17]: This was resolved in https://github.com/gatsbyjs/gatsby/pull/37933 but this PR need to be rebased or master merged in

Comment (pieh) [reply packages/gatsby-source-wordpress/src/steps/source-nodes/index.ts:17]: merged master in https://github.com/gatsbyjs/gatsby/pull/37920/commits/d2ea933ddc9801c783962c9b3e7e5095ca096283 so this is resolved now

File: packages/gatsby-source-wordpress/src/steps/source-nodes/update-nodes/wp-actions/update.js [hunk 2]
```diff
@@ -304,13 +304,17 @@ const wpActionUPDATE = async ({ helpers, wpAction }) => {
...
+      // by removing it from cached nodes so it's garbage collected by Gatsby
+      const validNodeIds = cachedNodeIds.filter(cachedId => cachedId !== nodeId)
 
-    await setPersistentCache({ key: CREATED_NODE_IDS, value: validNodeIds })
+      await setPersistentCache({ key: CREATED_NODE_IDS, value: validNodeIds })
+    }
 
     if (existingNode) {
+      // calling touchNode here ensures owners is populated before we delete.
+      // In some cases calling delete node without touching the node first throws an error. this is a bug in Gatsby that needs to be fixed
```
Comment (pieh) [inline packages/gatsby-source-wordpress/src/steps/source-nodes/update-nodes/wp-actions/update.js:317]: with persisted type owner tracking I do think we fixed the problem, but also we need the code still for gatsby versions before that fix - so mostly it's about comment text not being right anymore after https://github.com/gatsbyjs/gatsby/pull/37782 I think

**Comment by TylerBarnes** [comment]
Thanks @LekoArts , I forgot about that one. This is a cherry pick from that one of just the source-wp changes. I think I opened that draft just to run CI checks iirc. I'll close it now

**Comment by DanielRiveraHQ** [comment]
@TylerBarnes Let me know if I can help testing this with our Revere site

**Comment by TylerBarnes** [comment]
@DanielRiveraHQ that would be massively helpful!

You'll need `gatsby@5.9.0-touch-nodes-optout.49` and `gatsby-source-wordpress@7.9.0-stateful-wp.2`

**Review by pieh — APPROVED** [review]

**Pull request merged**